addons/transifex/models/transifex_code_translation.py

```diff
@@ -0,0 +1,162 @@
+            # transactions without table look can only see existing translations
+            pass
+
+        return {
+            'name': 'Code Translations',
+            'type': 'ir.actions.act_window',
+            'res_model': 'transifex.code.translation',
+            'view_mode': 'list',
+            'help': _('''<p class="o_view_nocontent_smiling_face">
+                            No Code Translation
+                        </p><p>
+                            Either the file for code translation is missing,
+                        </p><p>
+                            Or please change the condition for the filter
+                        </p><p>
+                            Or someone else is creating the code translation data and please do the action few seconds later
+                        </p>'''),
+        }
+
+    def _compute_transifex_url(self):
+        self.transifex_url = False
+        TransifexTranslation = self.env['transifex.translation']
+        transifex_metadata = TransifexTranslation._get_transifex_metadata(set(self.mapped('module')), list(set(self.mapped('lang'))))
+        TransifexTranslation._update_transifex_url(transifex_metadata, self)
+
+
+class TransifexTranslation(models.AbstractModel):
+    _name = "transifex.translation"
+    _description = "Transifex Translation"
+
+    def _get_transifex_metadata(self, modules, langs):
+        # e.g. 'https://www.transifex.com/odoo/'
+        base_url = self.env['ir.config_parameter'].sudo().get_param('transifex.project_url')
+        if not base_url:
+            return None
+        base_url = base_url.rstrip('/')
+
+        res_langs = self.env['res.lang'].search([('code', 'in', langs)])
+        lang_to_iso = dict((l.code, l.iso_code) for l in res_langs)
+        if not lang_to_iso:
+            return None
+
+        tx_config_file = ConfigParser()
+        tx_sections = []
+        for addon_path in odoo.addons.__path__:
+            tx_path = opj(addon_path, '.tx', 'config')
+            if isfile(tx_path):
+                tx_config_file.read(tx_path)
+                # first section is [main], after [odoo-16.sale]
+                tx_sections.extend(tx_config_file.sections()[1:])
+
+            # parent directory ad .tx/config is root directory in odoo/odoo
+            tx_path = opj(addon_path, pardir, '.tx', 'config')
+            if isfile(tx_path):
+                tx_config_file.read(tx_path)
+                tx_sections.extend(tx_config_file.sections()[1:])
+
+        if not tx_sections:
+            return None
+
+        base_url = base_url.rstrip('/')
+        # .tx/config files contains the project reference
+        # using ini files like '[odoo-master.website_sale]'
+        projects = {}
+        for module in modules:
+            for section in tx_sections:
+                tx_project, tx_mod = section.split('.')
+                if tx_mod == module:
+                    projects[module] = tx_project
```

Comment:
I hope the list `tx_sections` is small, otherwise this is a nasty O(n²) algorithm.